### PR TITLE
Consolidate commonly used GroupResource objects

### PR DESCRIPTION
**Pull request by nrb**
Cleans up some tech debt from #463
Signed-off-by: Nolan Brubaker <nolan@heptio.com>

### CODE REVIEW

File: pkg/kuberesource/kuberesource.go [hunk 1]
```diff
@@ -14,12 +14,16 @@ See the License for the specific language governing permissions and
...
+package kuberesource
 
 import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
-var PodsGroupResource = schema.GroupResource{Group: "", Resource: "pods"}
-var JobsGroupResource = schema.GroupResource{Group: "batch", Resource: "jobs"}
-var NamespacesGroupResource = schema.GroupResource{Group: "", Resource: "namespaces"}
+var Pod = schema.GroupResource{Group: "", Resource: "pods"}
```
Comment (ncdc) [inline pkg/kuberesource/kuberesource.go]: I think these should all be plural. WDYT @skriss ?

Comment (nrb) [reply pkg/kuberesource/kuberesource.go]: I actually started it that way, then got skeptical of myself.

I think it makes sense to be plural and match the actual `Resource` field.

Comment (skriss) [reply pkg/kuberesource/kuberesource.go]: plural sounds good to me.

File: pkg/kuberesource/kuberesource.go [hunk 2]
```diff
@@ -14,12 +14,16 @@ See the License for the specific language governing permissions and
...
+package kuberesource
 
 import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
-var PodsGroupResource = schema.GroupResource{Group: "", Resource: "pods"}
-var JobsGroupResource = schema.GroupResource{Group: "batch", Resource: "jobs"}
-var NamespacesGroupResource = schema.GroupResource{Group: "", Resource: "namespaces"}
+var Pods = schema.GroupResource{Group: "", Resource: "pods"}
```
Comment (skriss) [inline pkg/kuberesource/kuberesource.go]: formatting nit, but I'd (a) sort by group and (b) declare like:
```
var (
    Pods = ...
    Namespaces = ...
)
```
which gofmt will auto-align nicely.

File: pkg/restore/restore.go [hunk 3]
```diff
@@ -825,14 +826,12 @@ func hasControllerOwner(refs []metav1.OwnerReference) bool {
 
 // TODO(0.9): These are copied from backup/item_backupper. Definititions should be moved
```
Comment (skriss) [inline pkg/restore/restore.go]: Pls remove this comment :)

Comment (nrb) [reply pkg/restore/restore.go]: Nice catch, done.

**Comment by skriss** [comment]
LGTM.

**Pull request merged**